### PR TITLE
support ddl based table declare

### PR DESCRIPTION
**Pull request by chenqin**
prototype parse postgresSQL DDL create materialized view from data source

### CODE REVIEW

File: src/ingest/SpecRepo.cpp [hunk 1]
```diff
@@ -295,6 +304,293 @@ void SpecRepo::process(
...
+      "     topic='mytopic' and\n"
+      "     source='s3://nebula/ephemeral/dt={date}/downstream={ds}/contenttype={ct}/pinformat={pf}/eventtype={et}/part-r-{bucket}-fb7ea820-76a3-4c60-be79-956727df7593.gz.parquet' and\n"
+      "     schema='<Row:dict:string,host:string,tag:string,lang:double,stack:string>' and\n"
+      "     loader='Roll' and\n"
+      "     timestamp='provided' and\n"
+      "     batch=500 and\n"
+      "     topic_retention='90000' and\n"
+      "     protocol='binary' and"
+      "     time_partition='DAILY' and"
+      "     macro.et in (1,2,3) and\n"
```
Comment (chenqin) [inline src/ingest/SpecRepo.cpp:573]: @shawncao relates to adding star-partition macro in file path.

Comment (shawncao) [reply src/ingest/SpecRepo.cpp:573]: Thanks Chen, let me take a look tomorrow, feel down already, 🙁

Comment (chenqin) [reply src/ingest/SpecRepo.cpp]: refactor with tests running, removed hardcode, should helps with events/dt={date}/action_type={action} use case.

Comment (shawncao) [reply src/ingest/SpecRepo.cpp]: In this DDL statement, are you looking to compliant with ANSI SQLL, I haven't found an official definition for `create table definition`, but in my memory of Presto and SQL server, the ANSI SQL style of `create table statement` looks like this: 

```
CREATE TABLE <TBL_NAME>
(
    <col-name> <col-type>,
)
WITH (
   <all table properties> (structure of nested key-values)
)
```

Comment (shawncao) [reply src/ingest/SpecRepo.cpp]: BTW, for this DDL language support, what is your goal? why is it better comparing to plain YAML definition?

I'm wondering if we can separate the interfaces into
- keep YAML definition as YAML style
- introduce nebula console (cmd) and allow user to type SQL/DDL statement to interact with Nebula 

Looks like a fun stuff.

**Comment by chenqin** [comment]
I will be on vacation starting Monday. 
A few words on this massive pr.
- it interpret postgresSQL DDL and assign values to tablespec(s) 
- the logic is opt-in when user set table.ddl in yaml file otherwise fallback to YAML properties logic

File: src/ingest/SpecRepo.cpp (hunk 1 above)
Comment (chenqin) [reply src/ingest/SpecRepo.cpp]: https://github.com/varchar-io/nebula/issues/62 here is inspiring example I would like to achieve.

Comment (chenqin) [reply src/ingest/SpecRepo.cpp]: Yes, that's something we can consider, we need to map what nebula offers to what folks perceive.
- does nebula table support primary key partition
- does  nebula table support setting indexes over arbitrary column(s)

In another view, the #62 is more aligned with materialized view
https://stackoverflow.com/questions/4218657/materialized-view-vs-tables-what-are-the-advantages

File: src/ingest/SpecRepo.cpp [hunk 2]
```diff
@@ -295,6 +304,28 @@ void SpecRepo::process(
       table, version, buildIdentityByTime(table->timeSpec), table->name, 0, SpecState::NEW, 0));
     return;
   }
+  //hack, if table has sql statement, switch to sql parser
```
Comment (chenqin) [inline src/ingest/SpecRepo.cpp:307]: @shawncao I agree we should put this to another code path where we read from cmd or sqlscript in future.

**Comment by chenqin** [comment]
take back and prototype myself. 
- introduce cognitive burden to learn two things
- not clear how much benefits over yaml in current stage.

**Pull request closed without merging**